common/src/main/java/com/scottlogic/deg/common/profile/FieldWrapper.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.common.profile;
+
+public final class FieldWrapper<T> {
```

Comment:
Think we should give thisclass a better name if can.  Maybe a name that reflects what its menat to encpsulate.  Which is a Field and a field it depends on.  Also why is it generic.  I thought other member variable would always just be a Field (based on me finding usages of this class).  Or is it designed so it can be something else too?  (eg a hardcoded value or 'now'?)

Reply:
Originally I used it in multiple places, with different types. Over refactoring it has now reduced to one case. I'll remove the generic and replace it with a concrete type, which makes the naming more convenient and moves it out of the `common` module into `generator`.